taglib/toolkit/tfilestream.cpp

```diff
@@ -51,10 +51,18 @@ namespace
   {
     const DWORD access = readOnly ? GENERIC_READ : (GENERIC_READ | GENERIC_WRITE);
 
-    if(!path.wstr().empty())
-      return CreateFileW(path.wstr().c_str(), access, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
-    else if(!path.str().empty())
-      return CreateFileA(path.str().c_str(), access, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
+	if(!path.wstr().empty())
+#if (_WIN32_WINNT >= 0x0602)
+	  return CreateFile2(path.wstr().c_str(), access, FILE_SHARE_READ, OPEN_EXISTING, NULL);
+#else
+	  return CreateFileW(path.wstr().c_str(), access, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
+#endif
+	else if(!path.str().empty())
+#if (_WIN32_WINNT >= 0x0602)
+	  return CreateFile2(path.toString().toCWString(), access, FILE_SHARE_READ, OPEN_EXISTING, NULL);
+#else
+	  return CreateFileA(path.str().c_str(), access, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
+#endif
```

Comment:
Your code and mine are exactly equivalent, since `path.toString()` has the same branch as `if(!path.wstr().empty()) ...` in it. My intention is to keep our codebase maintainable by having simpler `#if ~ #endif` structures.